lib/scaffold.js

```diff
@@ -0,0 +1,23 @@
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+const mustache = require('mustache');
+
+// takes a .mustache file and renders this to a target folder with a given
+// filename and file extension
+module.exports = (mustacheOpts, basePath) => file => {
```

Comment:
I think have `scaffold` receive an `opts` object which could have:
- `opts.files (Array of files)`
- `opts.basePath`
- `opts.mustacheOpts`

And then move any of the file mapping/extend logic into this function.

Also maybe use a named function for better error handling?